pkg/solana/client/client.go

```diff
@@ -79,6 +102,56 @@ func NewClient(endpoint string, cfg config.Config, requestTimeout time.Duration,
 	}, nil
 }
 
+var _ mn.RPCClient[mn.StringID, *Head] = (*Client)(nil)
+var _ mn.SendTxRPCClient[*solana.Transaction] = (*Client)(nil)
+
+// TODO: BCI-4061: Implement Client for MultiNode
+
+func (c *Client) Dial(ctx context.Context) error {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c *Client) SubscribeToHeads(ctx context.Context) (<-chan *Head, mn.Subscription, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c *Client) SubscribeToFinalizedHeads(ctx context.Context) (<-chan *Head, mn.Subscription, error) {
+	//TODO implement me
+	panic("implement me")
+}
```

Comment:
does this require WS support?

i thought we were slowly moving away from it because it's not reliable? we never implemented WS support because it had issues with memory leaks and instead went with a polling mechanism

Reply:
This shouldn't require WS support, we can implement it with polling as well